app/services/wallet_transactions/create_service.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module WalletTransactions
+  class CreateService < BaseService
+    def create(**args)
+      return result unless valid?(**args)
+
+      handle_paid_credits(args[:wallet_id], args[:paid_credits]) if args[:paid_credits]
+      handle_granted_credits(args[:wallet_id], args[:granted_credits]) if args[:granted_credits]
+    end
+
+    private
+
+    def handle_paid_credits(wallet_id, paid_credits)
```

Comment:
What about using named parameters for `handle_paid_credits` and `handle_granted_credits`?

Reply:
This will be done on the other PR working on this service!